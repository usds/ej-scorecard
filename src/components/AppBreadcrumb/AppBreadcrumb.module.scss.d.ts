// declare namespace AppBreadcrumbNamespace {
//   export interface IAppBreadcrumbScss {
//     appBreadcrumbContainer: string;
//   }
// }

// declare const AppBreadcrumbScssModule: AppBreadcrumbNamespace.IAppBreadcrumbScss & {
//   /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
//   locals: AppBreadcrumbNamespace.IAppBreadcrumbScss;
// };

// export = AppBreadcrumbScssModule;
