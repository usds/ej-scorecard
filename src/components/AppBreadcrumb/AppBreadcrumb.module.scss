@use "uswds-core" as *;

.appBreadcrumbContainer{

}
