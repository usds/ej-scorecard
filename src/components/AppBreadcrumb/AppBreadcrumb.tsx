import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
} from '@trussworks/react-uswds';
import React from 'react';

// import * as styles from './AppBreadcrumb.module.scss';

import { AppBreadcrumbProps } from '@/types';
import { Link } from 'gatsby';

const AppBreadcrumb: React.FC<AppBreadcrumbProps> = ({ agencyName }) => {
  return (
    <BreadcrumbBar>
      <Breadcrumb>
        <BreadcrumbLink href="/scorecard">
          <Link to={`/scorecard`} activeClassName={`usa-current`}>
            Scorecard
          </Link>
        </BreadcrumbLink>
      </Breadcrumb>
      <Breadcrumb>{agencyName}</Breadcrumb>
    </BreadcrumbBar>
  );
};

export default AppBreadcrumb;
