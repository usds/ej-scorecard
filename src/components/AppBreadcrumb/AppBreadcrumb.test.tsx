import createComponentWithIntl from '@/test/testHelpers';

import AppBreadcrumb from './AppBreadcrumb';

describe(`rendering of AppBreadcrumb Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <AppBreadcrumb agencyName="TestAgency" />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
