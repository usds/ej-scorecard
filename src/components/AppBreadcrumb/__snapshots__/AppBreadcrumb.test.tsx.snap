// Vitest Snapshot v1

exports[`rendering of AppBreadcrumb Component > checks if component renders 1`] = `
<nav
  aria-label="Breadcrumbs"
  className="usa-breadcrumb"
>
  <ol
    className="usa-breadcrumb__list"
  >
    <li
      className="usa-breadcrumb__list-item"
    >
      <a
        className="usa-breadcrumb__link"
        href="/scorecard"
      >
        <a
          activeClassName="usa-current"
          href="/scorecard"
        >
          Scorecard
        </a>
      </a>
    </li>
    <li
      className="usa-breadcrumb__list-item"
    >
      TestAgency
    </li>
  </ol>
</nav>
`;
